src/systems/systems/src/navigation/radio_altimeter.rs

```diff
@@ -0,0 +1,864 @@
+    const MAXIMUM_POWER_HOLDOVER: u64 = 35_000;
+
+    pub fn new(
+        context: &mut InitContext,
+        number: usize,
+        aircraft_installation_delay: Ala52BAircraftInstallationDelay,
+        powered_by: ElectricalBusType,
+    ) -> Self {
+        let was_powered = context.has_engines_running();
+        Self {
+            number,
+            aircraft_installation_delay,
+            powered_by,
+            is_powered: false,
+            power_holdover: Duration::from_secs_f64(random_from_range(
+                Self::MINIMUM_POWER_HOLDOVER as f64 / 1000.,
+                Self::MAXIMUM_POWER_HOLDOVER as f64 / 1000.,
+            )),
+            unpowered_for: if was_powered {
+                Duration::ZERO
+            } else {
+                Duration::from_millis(Self::MAXIMUM_POWER_HOLDOVER)
+            },
+            self_check_time: Duration::from_secs_f64(random_from_range(
+                Self::MINIMUM_STARTUP_TIME_MILLIS as f64 / 1000.,
+                Self::MAXIMUM_STARTUP_TIME_MILLIS as f64 / 1000.,
+            )),
+
+            runtime: if was_powered {
+                Some(Ala52BRadioAltimeterRuntime::new_running(
+                    aircraft_installation_delay,
+                ))
+            } else {
+                None
+            },
+            failure: Failure::new(FailureType::RadioAltimeter(number)),
+            radio_altitude_id: context.get_identifier(Self::radio_altitude_id(number)),
+        }
+    }
+
+    fn radio_altitude_id(number: usize) -> String {
+        format!("RA_{}_RADIO_ALTITUDE", number)
+    }
+
+    pub fn update(&mut self, context: &UpdateContext, transceivers: &Ala52BTransceiverPair) {
+        if self.is_powered {
+            self.unpowered_for = Duration::ZERO;
+        } else {
+            self.unpowered_for += context.delta();
+        }
+
+        // Check if the internal state (runtime) is lost because either the unit failed, or the
+        // power holdover has been exceed
+        if self.failure.is_active() || self.unpowered_for > self.power_holdover {
+            // Throw away the simulated software runtime
+            self.runtime = None;
+            return;
+        }
+
+        // As long as we're powered, we can proceed normally. If not, we can't run the runtime, but
+        // it's state will be frozen and if power is restored soon enough, we can proceed
+        // immediately without waiting for the runtime to start up again.
+        if self.is_powered {
+            // Either initialize and run or continue running the existing runtime
+            let self_check = self.self_check_time;
+            let aid = self.aircraft_installation_delay;
+            let runtime = self
+                .runtime
+                .get_or_insert_with(|| Ala52BRadioAltimeterRuntime::new(self_check, aid));
+            runtime.update(context, transceivers);
+        }
+    }
+
+    pub fn number(&self) -> usize {
+        self.number
+    }
```

Comment:
```suggestion
```

Reply:
I added this and kept it for now because the RA does in fact know which RA it is, thanks to pin-programming.
This is especially important later on because the RA would set the SDI bits (Source Destination Identifier) in any ARINC words based on whether it's RA 1 or 2 (or 3 for the A380). If we drop this fn, then the `number` field becomes unused, and it seemed nicer to explicitly make it part of the external API rather than `#[allow(dead_code)]` the field above.

Thoughts?

Reply:
If you expect usages outside of the module, then I'd leave it in. If you only expect usages inside the module, I'd add `#[allow(dead_code)]` for now.

Reply:
Alright, then to keep the API clean I'll just go with allowing the dead code. Thanks for your input.